File: app/src/main/java/farasense/mobile/view/components/FaraSenseWrappingViewPager.java
package farasense.mobile.view.components;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.util.TypedValue;
import android.view.View;

public class FaraSenseWrappingViewPager extends ViewPager {

    private int height;

    public FaraSenseWrappingViewPager(Context context) {
        super(context);
        height = 0;
    }

    public FaraSenseWrappingViewPager(Context context, AttributeSet attrs){
        super(context, attrs);
        height = 0;
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {

        int children = getChildCount();
        for(int i = 0; i < children; i++) {
            View child = getChildAt(i);
            child.measure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED));
            int h = child.getMeasuredHeight();
            if(h > height) {
                height = h;
            }
        }

        if (height != 0) {
            heightMeasureSpec = MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY);
        }

        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    }
}

File: app/src/main/java/old_farasense/view/components/FaraSenseWrappingViewPager.kt
package old_farasense.view.components

import android.content.Context

import android.util.AttributeSet
import android.view.View
import androidx.viewpager.widget.ViewPager

class FaraSenseWrappingViewPager : ViewPager {
    private var viewHeight: Int = 0

    constructor(context: Context) : super(context) {
        viewHeight = 0
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        viewHeight = 0
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        var heightMeasureSpec = heightMeasureSpec

        val children = childCount
        for (i in 0 until children) {
            val child = getChildAt(i)
            child.measure(widthMeasureSpec, View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED))
            val h = child.measuredHeight
            if (h > viewHeight) {
                viewHeight = h
            }
        }

        if (viewHeight != 0) {
            heightMeasureSpec = View.MeasureSpec.makeMeasureSpec(viewHeight, View.MeasureSpec.EXACTLY)
        }

        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
    }
}